src/hyphenClient.ts

```diff
@@ -0,0 +1,67 @@
+import { EvaluationResponse, HyphenEvaluationContext, TelemetryPayload } from './types';
+import { horizonEndpoints } from './config';
+import { Logger } from '@openfeature/web-sdk';
+
+export class HyphenClient {
+  private readonly publicKey: string;
+  private readonly horizonServerUrls: string[];
+
+  constructor(publicKey: string, horizonServerUrls: string[] = []) {
+    this.publicKey = publicKey;
+    this.horizonServerUrls = [...horizonServerUrls, horizonEndpoints.evaluate];
+  }
+
+  async evaluate(context: HyphenEvaluationContext, logger?: Logger): Promise<EvaluationResponse> {
+    return await this.fetchEvaluationResponse(this.horizonServerUrls, context, logger);
+  }
+
+  async postTelemetry(payload: TelemetryPayload, logger?: Logger) {
+    await this.httpPost(horizonEndpoints.telemetry, payload, logger);
+  }
+
+  private async fetchEvaluationResponse(
+    serverUrls: string[],
+    context: HyphenEvaluationContext,
+    logger?: Logger,
+  ): Promise<EvaluationResponse> {
+    let lastError: unknown;
+
+    for (const url of serverUrls) {
+      try {
+        const response = await this.httpPost(url, context, logger);
+        return await response.json();
+      } catch (error) {
+        lastError = error;
+        logger?.debug('Failed to fetch evaluation: ', url, error);
```

Comment:
between here and the httpPost help every error will get logged twice. We should only do this once